app/controllers/price_simulations_controller.rb

```diff
@@ -1,6 +1,16 @@
 class PriceSimulationsController < ApplicationController
   def compute
-    # FIXME at iteration 4
-    render json: { price: 0 }
+    p = params.permit(:kind, :title, :isbn, :purchase_price, :is_hot, :width, :height, :source, :format, :duration, :quality).to_h
```

Comment:
Did you consider factorizing this code with the equivalent logic in the parsers (like [this one](https://github.com/joboccara/my-media-store/pull/6/files#diff-5cc4d3d3a57553b8f3f4307bf463c7c61d6dbbfd2dcf6460d75323c3e2adaf97R24) for example)?

Reply:
I don't like this line but didn't had a much better idea to improve.
Having a `allowed_params` method returning a list of permitted key on each pricer would be a half solution IMO because of coupling. It assumes you don't have nested values.
A better one would be a real parser that could be combined, but much more work there ;)